src/migration/1714662136871-SquadTotalUpvotesTrigger.ts

```diff
@@ -0,0 +1,47 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SquadTotalUpvotesTrigger1714662136871
+  implements MigrationInterface
+{
+  name = 'SquadTotalUpvotesTrigger1714662136871';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION update_squad_upvotes_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          IF NEW.upvotes <> OLD.upvotes THEN
+            UPDATE  source
+            SET     flags = jsonb_set(
+                              flags,
+                              '{totalUpvotes}',
+                              to_jsonb(
+                                GREATEST(
+                                  0,
+                                  COALESCE(CAST(flags->>'totalUpvotes' AS INTEGER), 0) +
+                                  (CASE WHEN NEW.upvotes > OLD.upvotes THEN 1 ELSE -1 END)
```

Comment:
But you don't know if upvotes for how much the upvotes changed. For example upvotes could change from 8 to 10. You will increment by 1, when in reality it should add two upvotes.

Reply:
See my comment https://github.com/dailydotdev/daily-api/pull/1866#discussion_r1588928823

Reply:
Would it? Do we not send an upvote one at a time per user and they will run on their own even if two users hit the upvote at the same time?

Reply:
Most of the time yes, and if you listened to `user_post` that would be the case since each new row means upvote count changed by at most 1 (+ or -). But since you are looking at accumulator on post you have to take it into account.

Reply:
Between I think in this trigger you can actually use `WHEN NEW.upvotes <> OLD.upvotes` to trigger only when upvotes count change.

Reply:
Oh, nice. Didn't know you can actually specify that.